### PR TITLE
Futures compatibility

### PR DESCRIPTION
**Pull request by SemVerTsar**
This builds upon @Protosac's work on adding python2 compatibility.

@incuna/backend Please review

### CODE REVIEW

**Comment by KevinEtienne** [comment]
Looks good to me

File: tests/test_fields.py [hunk 1]
```diff
@@ -90,16 +108,16 @@ def test_value_returned_is_not_bytea(self):
         EncryptedModelFactory.create()
 
         instance = self.model.objects.get()
-        self.assertIsInstance(instance.digest_field, str)
-        self.assertIsInstance(instance.hmac_field, str)
+        self.assertIsInstance(instance.digest_field, unicode)
+        self.assertIsInstance(instance.hmac_field, unicode)
```
Comment (adam-thomas) [inline tests/test_fields.py]: Does this still work on Python 3?

File: tests/test_fields.py [hunk 2]
```diff
@@ -21,12 +23,28 @@
...
+class TestUtilsMixin(object):
+    def assertCountEqual(self, a, b):
+        """Assert a and b are equal in length."""
+        return len(a) == len(b)
+
+
+@contextmanager
+def sub_test(field):
+    """Implement sub_test."""
+    yield
```
Comment (adam-thomas) [inline tests/test_fields.py]: This can't be all that subTest does, surely?

Comment (meshy) [inline tests/test_fields.py]: How does this work? `self.subTest` adds the ability for a test to fail once per iteration in python 3. Does that still work with this?

File: tests/test_fields.py (hunk 1 above)
Comment (NoLogo) [inline tests/test_fields.py]: Yes, on line 26-27
```
if PY3:
    unicode = str
```

File: tests/test_fields.py (hunk 2 above)
Comment (NoLogo) [inline tests/test_fields.py]: That was added by @protosac from what I can tell it works, but I'm happy to remove if it's easier.

**Comment by NoLogo** [comment]
Just like old times

File: tests/test_fields.py (hunk 2 above)
Comment (meshy) [inline tests/test_fields.py]: I'd be interested to see how it behaves where it fails. On python 3 I'd expect to see a traceback for each iteration.

File: tests/test_fields.py (hunk 1 above)
Comment (meshy) [inline tests/test_fields.py]: Might be better to take the "python 3 first" approach, and stick with `str`. Alternatively, how about something that can't be confused: `text_type` vs `bytes_type`, or similar. I think that's what `six` does, no?

**Comment by KevinEtienne** [comment]
tenor

**Comment by adam-thomas** [comment]
http://i.imgur.com/0QVN6k9.mp4

**Comment by peterfarrell** [comment]
Given that the last version of Django (1.11) to support Python 2 reaches EOL in January of 2020, I'm hesitant to vote to accept this PR. We're looking at 14 months before Python 2 is no longer applicable for Django development.

**Comment by SemVerTsar** [comment]
I'd say that's fair. This work was started in April 2017 & this PR has had no activity for over a year. Shall I close it?

**Comment by peterfarrell** [comment]
@SemVerTsar Yea, I didn't want to close it without ceremony. I'll close it out now.

**Pull request closed without merging**